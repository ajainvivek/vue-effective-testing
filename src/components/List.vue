<template>
  <div class="List">
    <ul
        v-infinite-scroll="fetchMore"
        infinite-scroll-disabled="loading"
        infinite-scroll-distance="10"
        class="List__list"
    >
        <li class="List__listitem" v-for="item in list" :key="item">{{ item }}</li>
    </ul>
    <p v-show="loading" class="List__listitem-loading">
        <mt-spinner type="fading-circle" class="spinner"></mt-spinner>
        Loading...
    </p>
   </div> 
</template>

<style>
    .mint-spinner-fading-circle {
        display: inline-block;
        vertical-align: middle;
        margin-right: 5px;
    }
</style>

<style scoped>
    .List {
        margin-top: 40px;
    }

    .List__list {
        margin: 0;
        padding: 0;
    }

    .List__listitem {
        height: 70px;
        line-height: 70px;
        border-bottom: solid 1px #eee;
        text-align: center;
        list-style: none;
    }

    .List__listitem-loading {
        text-align: center;
        height: 50px;
        line-height: 50px;
        margin: 0;
    }
</style>

<script>
export default {
  name: 'list',
  props: [
    'loading',
    'list'
  ],
  methods: {
    fetchMore () {
      this.$emit('fetchMore');
    }
  }
}
</script>
