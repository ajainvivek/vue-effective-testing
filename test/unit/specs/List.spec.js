import { shallow } from '@vue/test-utils'
import List from '@/components/List.vue'

describe('List.vue', () => {
  
  let list = [];

  beforeEach(() => {
    list = [{
      id: 1,
      title: "Luke Skywalker"
    }, {
      id: 2,
      title: "C-3PO"
    }];
  });

  it('should render EMPTY list item', () => {
    list = []
    const wrapper = shallow(List, {
      propsData: { list }
    })
    expect(wrapper.findAll('li')).toHaveLength(list.length)
  });

  it('should render list item', () => {
    const wrapper = shallow(List, {
      propsData: { list }
    })
    expect(wrapper.findAll('li')).toHaveLength(list.length)
  });

  it('matches snapshot', () => {
    const wrapper = shallow(List, {
      propsData: { list }
    })
    expect(wrapper.html()).toMatchSnapshot()
  });
})
