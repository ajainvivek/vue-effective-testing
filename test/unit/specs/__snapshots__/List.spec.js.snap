// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue matches snapshot 1`] = `
<div class="List">
    <ul class="List__list">
        <li class="List__listitem">Luke Skywalker</li>
        <li class="List__listitem">C-3PO</li>
    </ul>
    <p class="List__listitem-loading" style="display: none;">
        Loading...
    </p>
</div>
`;
